src/__tests__/union.test.ts

```diff
@@ -76,6 +76,53 @@ describe('union', () => {
         });
       });
 
+      describe('readable/writable', () => {
+        it('writes to the last vol added', () => {
+          const vol1 = Volume.fromJSON({});
+          const vol2 = Volume.fromJSON({});
+          const ufs = new Union();
+          ufs.use(vol1 as any).use(vol2 as any);
+          ufs.writeFileSync('/foo', 'bar');
+          expect(vol2.readFileSync('/foo', 'utf8')).toEqual('bar');
+        });
+
+        it('writes to the latest added volume with writable=false vol', () => {
+          const vol1 = Volume.fromJSON({});
+          const vol2 = Volume.fromJSON({});
+          const ufs = new Union();
+          ufs.use(vol1 as any).use(vol2 as any, { writable: false });
+          ufs.writeFileSync('/foo', 'bar');
+          expect(vol1.readFileSync('/foo', 'utf8')).toEqual('bar');
+        });
+
+        it('writes to the latest added writable vol', () => {
+          const vol1 = Volume.fromJSON({});
+          const vol2 = Volume.fromJSON({});
+          const ufs = new Union();
+          ufs.use(vol1 as any).use(vol2 as any, { readable: false });
+          ufs.writeFileSync('/foo', 'bar');
+          expect(vol2.readFileSync('/foo', 'utf8')).toEqual('bar');
+        });
+
+        it('throw error if write operation attempted with all volumes writable=false', () => {
```

Comment:
```suggestion
        it('throws an error when calling a write method', () => {
```

Reply:
Could you encapsulate these in `describe`s?

i.e

````
describe('when none of the volumes are writable', () => {
  ...
});

describe('when none of the volumes are readable', () => {
  ...
});
```

Reply:
outside of the top-level sync/async/promise describes? There's only one such test for each...

Reply:
Describes help to express particular concepts & aspects of a test, by providing a way to encapsulate said concepts & aspects in a block.

This is useful even if there is only one test, because we might add more tests in the future, and they aid in discovery.

A good rule of thumb for when it *might* be useful to use a describe is when you find yourself writing test cases that contain conditionals in their title, especially if they're lengthly:

* it does this *if* that is true
* it sorts *when* the array is empty
* test that an error is not thrown if the array is empty when calling the sort method
* it errors on the first test *with* the failFast option set to true

It is possible to take this to an extreme of course, and isn't always right but generally if you're using both a conditional *and* a long test title then using a describe is the right move.

To use this test title as an example:

"it throw error *if* write operation attempted *with* all volumes writable=false"

As a single test case that is reasonably long, and "what happens when none of the volumes are writable?" is a question that comes naturally.

Hence, breaking it into:

* when none of the volumes are writable:
  * it throws an error when calling a write method

Helps to improve discovery and maintainability.

I'm also happy if you want to chuck a inverse test with each of those, i.e

```
* when none of the volumes are readable:
  * it does not throw when calling a write method
  * it throws an error when calling a read method

* when none of the volumes are writable:
  * it throws an error when calling a write method
  * it does not throw when calling a read method
```

Having symmetrical tests is useful as it describes the *expected behaviour* in the situation, which can help provide peace of mind, and acts as a sort of sanity check. And, this is all with almost-zero cost: these tests are boring, fast, and very similar meaning they come with minimal maintenance burden :)

--------

Just so we're def on the same page, I'm meaning keep things in the current describes, but create a new one at each of the current test sites, for the same reason:

```
* describe: sync methods
  * ... tests ...
  * describe: when none of the volumes are writable
     * ... tests ...
  * describe: when none of the volumes are readable
     * ... tests ...
* describe: async methods
  * ... tests ...
  * describe: when none of the volumes are writable
     * ... tests ...
  * describe: when none of the volumes are readable
     * ... tests ...
* describe: promise methods
  * ... tests ...
  * describe: when none of the volumes are writable
     * ... tests ...
  * describe: when none of the volumes are readable
     * ... tests ...
```

It's very likely that the surrounding tests don't follow this structure in quite the same way - that's fine; we can always improve existing code later.

What we're striving for is a good balance between being efficient, and improving as we go :)

Reply:
new test hierarchy added